crates/swc_ecma_minifier/tests/fixture/issues/6957/number-tostring/output.js

```diff
@@ -0,0 +1,291 @@
+assertEquals("NaN", NaN.toString());
+assertEquals("Infinity", (1 / 0).toString());
+assertEquals("-Infinity", (-1 / 0).toString());
+assertEquals("0", "0");
+assertEquals("9", "9");
+assertEquals("90", "90");
+assertEquals("90.12", "90.12");
+assertEquals("0.1", "0.1");
+assertEquals("0.01", "0.01");
+assertEquals("0.0123", "0.0123");
+assertEquals("111111111111111110000", "111111111111111110000");
+assertEquals("1.1111111111111111e+21", "1.1111111111111111e+21");
+assertEquals("1.1111111111111111e+22", "1.1111111111111111e+22");
+assertEquals("0.00001", "0.00001");
+assertEquals("0.000001", "0.000001");
+assertEquals("1e-7", "1e-7");
+assertEquals("1.2e-7", "1.2e-7");
+assertEquals("1.23e-7", "1.23e-7");
+assertEquals("1e-8", "1e-8");
+assertEquals("1.2e-8", "1.2e-8");
+assertEquals("1.23e-8", "1.23e-8");
+assertEquals("0", "0");
+assertEquals("-9", "-9");
+assertEquals("-90", "-90");
+assertEquals("-90.12", "-90.12");
+assertEquals("-0.1", "-0.1");
+assertEquals("-0.01", "-0.01");
+assertEquals("-0.0123", "-0.0123");
+assertEquals("-111111111111111110000", "-111111111111111110000");
+assertEquals("-1.1111111111111111e+21", "-1.1111111111111111e+21");
+assertEquals("-1.1111111111111111e+22", "-1.1111111111111111e+22");
+assertEquals("-0.00001", "-0.00001");
+assertEquals("-0.000001", "-0.000001");
+assertEquals("-1e-7", "-1e-7");
+assertEquals("-1.2e-7", "-1.2e-7");
+assertEquals("-1.23e-7", "-1.23e-7");
+assertEquals("-1e-8", "-1e-8");
+assertEquals("-1.2e-8", "-1.2e-8");
+assertEquals("-1.23e-8", "-1.23e-8");
+assertEquals("NaN", NaN.toString(16));
+assertEquals("Infinity", (1 / 0).toString(16));
+assertEquals("-Infinity", (-1 / 0).toString(16));
+assertEquals("0", "0");
+assertEquals("9", "9");
+assertEquals("5a", "5a");
+assertEquals("5a.1eb851eb852", 90.12.toString(16));
+assertEquals("0.1999999999999a", 0.1.toString(16));
+assertEquals("0.028f5c28f5c28f6", 0.01.toString(16));
+assertEquals("0.032617c1bda511a", 0.0123.toString(16));
+assertEquals("605f9f6dd18bc8000", "605f9f6dd18bc8000");
+assertEquals("3c3bc3a4a2f75c0000", "3c3bc3a4a2f75c0000");
+assertEquals("25a55a46e5da9a00000", "25a55a46e5da9a00000");
+assertEquals("0.0000a7c5ac471b4788", 0.00001.toString(16));
+assertEquals("0.000010c6f7a0b5ed8d", 0.000001.toString(16));
+assertEquals("0.000001ad7f29abcaf48", 0.0000001.toString(16));
+assertEquals("0.000002036565348d256", 0.00000012.toString(16));
+assertEquals("0.0000021047ee22aa466", 0.000000123.toString(16));
+assertEquals("0.0000002af31dc4611874", 0.00000001.toString(16));
+assertEquals("0.000000338a23b87483be", 0.000000012.toString(16));
+assertEquals("0.00000034d3fe36aaa0a2", 0.0000000123.toString(16));
+assertEquals("0", "0");
+assertEquals("-9", "-9");
+assertEquals("-5a", "-5a");
+assertEquals("-5a.1eb851eb852", -90.12.toString(16));
```

Comment:
This seems wrong.

Reply:
I did
```js
function assertEquals(a, b) {
    if (a !== b) {
        throw `${a} !== ${b}`
    }    
}
```

Reply:
Weird, Chrome tells me the result is NaN.

Reply:
I found the problem.

```js
-90.12.toString(16)
```

is different from 
```js
(-90.12).toString(16)
```

Reply:
Seems like `-` is processed after evaluating `.toString(16)`

Reply:
Can you fix the fixer, too?

Reply:
> Can you fix the fixer, too?

Sure

Reply:
@kdy1 The root cause :
https://github.com/swc-project/swc/blob/ac9687af5f2e66499e7df9c0ac2f547e4494bd29/crates/swc_ecma_transforms_optimization/src/simplify/expr/mod.rs#L836-L843

optimize_unary_expr will convert 
`UnaryExpr(-, Lit::Num(90.12))` into `Lit::Num(-90.12)`.

Where should we fix it? Disable this behavior or fix it in `fixer`?

Reply:
I think disabling this behavior would be better, as it's sort of `invalid` state